From ff1d69edc33c1b77cb48122116bd65c99fd06793 Mon Sep 17 00:00:00 2001
From: Yunlei Ding <yunlei.ding@citrix.com>
Date: Tue, 11 Mar 2014 09:49:59 +0000
Subject: [PATCH 04/11] virtio-blk: correctly link new request in virtio_blk_load()
 22739
To: andrew.cooper3@citrix.com

s->rq should be set with req instead of req-next.

Signed-off-by: Yunlei Ding <yunlei.ding@citrix.com>
diff --git a/hw/virtio-blk.c b/hw/virtio-blk.c
index fcf893a..f3a81e3 100644
--- a/hw/virtio-blk.c
+++ b/hw/virtio-blk.c
@@ -288,7 +288,7 @@ static int virtio_blk_load(QEMUFile *f, void *opaque, int version_id)
         VirtIOBlockReq *req = virtio_blk_alloc_request(s);
         qemu_get_buffer(f, (unsigned char*)&req->elem, sizeof(req->elem));
         req->next = s->rq;
-        s->rq = req->next;
+        s->rq = req;
     }
 
     return 0;
